test/array.test.rb

```diff
@@ -1,5 +1,25 @@
 # frozen_string_literal: true
 
+include Literal::Types
+
+test "===" do
+	assert Literal::Array(_Boolean) === Literal::Array(true).new(true)
+	assert Literal::Array(Object) === Literal::Array(Integer).new(1)
+	assert Literal::Array(Numeric) === Literal::Array(Integer).new(1)
+	assert Literal::Array(Numeric) === Literal::Array(Float).new(1.0)
+
+	assert Literal::Array(
+		Literal::Array(Numeric)
+	) === Literal::Array(
+		Literal::Array(Integer)
+	).new(
+		Literal::Array(Integer).new(1)
+	)
```

Comment:
Bit of a contrived example. You wouldn’t generally write code like this, but you could end up with a structure like this and Literal should handle it correctly.